[CI] SearchableSnapshotsPrewarmingIntegTests.testConcurrentPrewarming fails on CI

**Build scan**:
https://gradle-enterprise.elastic.co/s/7lytl32v5nhso
https://gradle-enterprise.elastic.co/s/7hdrmguklhuxq

**Repro line**:


**Reproduces locally?**:
No

**Applicable branches**:
master

**Failure excerpt**:
```
java.lang.AssertionError: Blob [__hBbReAqXTUGYJ6P1sR9ZTw.part7] not fully warmed
Expected: a value equal to or greater than <772L>
     but: <725L> was less than <772L>
	at __randomizedtesting.SeedInfo.seed([309FAA2B385BE603:78FC385D5105D25C]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsPrewarmingIntegTests.testConcurrentPrewarming(SearchableSnapshotsPrewarmingIntegTests.java:300)
```

--- a/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsPrewarmingIntegTests.java
+++ b/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsPrewarmingIntegTests.java
@@ -399,7 +399,7 @@ public class SearchableSnapshotsPrewarmingIntegTests extends ESSingleNodeTestCas
 
                     @Override
                     public void close() throws IOException {
-                        files.put(blobName, bytesRead);
+                        files.merge(blobName, bytesRead, Math::addExact);
                         super.close();
                     }
                 };

